### DISCUSSION TITLE
Running StereoKit on Oculus Quest 2

### DISCUSSION THREAD

**Discussion by jackdaus** [StereoKit/StereoKit#394] [Q&A] [answered]
Is there a way to run a StereoKit app on an Oculus Quest 2 without using the XPlat Template (i.e. https://github.com/StereoKit/SKTemplate-Universal)? 

I saw that in this repo (https://github.com/maluoi/SKDesktopDuplicator) that you are using the PassthroughFBExt _without_ Android specific code. This leads me to think it's possible. I'm interested in this because I think it will be easier for me to prototype using the simplified .NET template compared to using the more complicated XPlat template. 

A pointer in the right direction would be great!

**Comment by maluoi** [comment, marked as answer]
SKDesktopDuplicator was developed with Quest + Link, hand tracking and the FB passthough extension work quite nicely there. The actual desktop duplication code doesn't make much sense running directly on Quest, since Quest doesn't have any desktop to duplicate!

However, I have been working on a MAUI based template that should be moderately simpler than the Universal template. It's not finished quite yet, but is probably a soon thing. It'll still be the same architecturally, but will have a nicer project layout and creation process! Alternatively, you could have _just_ the Xamarin part of the Universal template, but that would be sacrificing desktop simulator support, since Xamarin becomes Android only. This never made quite as much sense to me, so I've never shipped that particular format as a template, but it's possible.

**Reply by jackdaus** [reply]
Thank you! I haven't used Quest + Link yet, so I'll experiment around with that. 

Looking forward to the MAUI template!

**Reply by maluoi** [reply]
Quest+Link is really great, and I do the majority of my in-headset development this way right now! I've got a nice long USB C cable connected to the Quest one way or another, doing Link just gives me better access to debugging tools and faster builds :)

If hands/passthrough don't work right away for you, you may need to enable them in the Oculus Beta settings. They're pretty recent!

![image](https://user-images.githubusercontent.com/8484884/173482522-26be66d4-be9a-449a-a54a-34d076c032db.png)
